third_party/proxy/util/utils.go

```diff
@@ -206,13 +206,30 @@ func MapCIDRsByIPFamily(cidrStrings []string) map[v1.IPFamily][]string {
 	return ipFamilyMap
 }
 
-func getIPFamilyFromCIDR(cidrStr string) (v1.IPFamily, error) {
-	_, netCIDR, err := net.ParseCIDR(cidrStr)
-	if err != nil {
-		return "", ErrAddressNotAllowed
+// GetIPFamilyFromIP Returns the IP family of ipStr, or IPFamilyUnknown if ipStr can't be parsed as an IP
+func GetIPFamilyFromIP(ipStr string) v1.IPFamily {
+	return convertToV1IPFamily(utilnet.IPFamilyOfString(ipStr))
+}
+
+func getIPFamilyFromCIDR(cidrStr string) v1.IPFamily {
+	return convertToV1IPFamily(utilnet.IPFamilyOfCIDRString(cidrStr))
+}
+
+// Convert netutils.IPFamily to v1.IPFamily
+func convertToV1IPFamily(ipFamily utilnet.IPFamily) v1.IPFamily {
+	switch ipFamily {
+	case utilnet.IPv4:
+		return v1.IPv4Protocol
+	case utilnet.IPv6:
+		return v1.IPv6Protocol
 	}
-	if utilnet.IsIPv6CIDR(netCIDR) {
-		return v1.IPv6Protocol, nil
+
+	return v1.IPFamilyUnknown
+}
+
+func IsVIPMode(ing v1.LoadBalancerIngress) bool {
+	if ing.IPMode == nil {
```

Comment:
In the K8s proxy code, they also check the FeatureGate: https://github.com/kubernetes/kubernetes/blob/4b8e819355d791d96b7e9d9efe4cbafae2311c88/pkg/proxy/util/utils.go#L335-L337

I guess in our case we believe there is no need to introduce support for that FeatureGate and it would just add complexity?

To be honest, I am not entirely sure why they have that check in the K8s code. It feels like extra redundancy in that case, since the apiserver already has that FeatureGate and will drop the field if it is not enabled. Maybe @tnqn knows.

Reply:
> I guess in our case we believe there is no need to introduce support for that FeatureGate and it would just add complexity?

You are right. There are some reasons why I removed the FeatureGate:

1. If a user wants to set an ingress IP to be used in `LoadBalancerIPModeProxy` (the related Service traffic is sent to external loadBalancer), personally, I think there is no need to set another global switch like the FeatureGate in kube-proxy. The users should know what they are doing.
2. The field associated with `LoadBalancerIPModeProxy` and `LoadBalancerIPModeVIP` has been in GA stage. IMO, I think the FeatureGate is redundant.
3. A not very strong reason, the change of this PR only affects the ingress IPs consumed by AntreaProxy. It doesn't change any code in AntreaProxy directly at all.

Reply:
> In some production clusters, there may be a version skew between kube-apiserver and kube-proxy

That makes sense. I believe kube-proxy is usually updated last too.

@hongliangl 
> The field associated with LoadBalancerIPModeProxy and LoadBalancerIPModeVIP has been in GA stage. IMO, I think the FeatureGate is redundant.

What do you mean by that?

In any case, I am fine with not having a FeatureGate in the Agent for this, as it does seem very low risk.

Reply:
@antoninbas I meant that the `LoadBalancerIPMode` has been supported in "k8s.io/api/core/v1" 0.29 which is in GA stage, but the latest kube-proxy code consuming `LoadBalancerIPMode` is still isolated by a FeatureGate, so I thought of that the FeatureGate is redundant. Thanks for @tnqn 's explanation, I understood the reason why the FeatureGate is added.

Reply:
@hongliangl I think what confused us is why you say it's in GA stage, regardless of K8s 1.29 or "k8s.io/api/core/v1" 0.29 (actually feature gate should have nothing to do with the package). It's alpha from what I can see: https://github.com/kubernetes/kubernetes/blob/20d0ab7ae808aaddb1556c3c38ca0607663c50ac/pkg/features/kube_features.go#L941-L944